app/server/api/search_api.py

```diff
@@ -28,19 +29,87 @@ def get(self):
+        # HANDLE PARAM : search_stirng
+        search_string = request.args.get('search_string') or ''      
 
+        # HANDLE PARAM : search_type
         # Valid search types are: `transfer_accounts` and `credit_transfers`
+        # Default: transfer_accounts
         search_type = request.args.get('search_type') or 'transfer_accounts'
         if search_type not in ['transfer_accounts', 'credit_transfers']:
             response_object = {
                 'message': 'Invalid search_type \'{}\'. Please use type \'transfer_accounts\' or \'credit_transfers\''.format(search_type),
             }
             return make_response(jsonify(response_object)), 400
 
+        # HANDLE PARAM : sort_by
+        # Valid params differ depending on sort_by. See: sort_by
+        # Default: rank
+        
+        # Aliases used for joining the separate sender and recipient objects to transfers
+        sender = aliased(User)
+        recipient = aliased(User)
+        # Build order by object
+        sort_types_to_database_types = {'first_name': User.first_name,
+                                        'last_name': User.last_name,
+                                        'email': User.email,
+                                        'date_account_created': User.created,
+                                        'rank': 'rank',
+                                        'balance': TransferAccount._balance_wei,
+                                        'status': TransferAccount.is_approved,
+                                        'amount': CreditTransfer.transfer_amount,
+                                        'transfer_type': CreditTransfer.transfer_type,
+                                        'approval': CreditTransfer.transfer_status,
+                                        'date_transaction_created': CreditTransfer.resolved_date,
+                                        'sender_first_name': sender.first_name,
+                                        'sender_last_name': sender.last_name,
+                                        'sender_email': sender.email,
+                                        'sender_date_account_created': recipient.created,
+                                        'recipient_first_name': recipient.first_name,
+                                        'recipient_last_name': recipient.last_name,
+                                        'recipient_email': recipient.email,
+                                        'recipient_date_account_created': recipient.created
+                                        }
+
+        # These lists are to validate the user input-- not using sort_types_to_database_types since credit_transfers and transfer_accounts have unique options 
+        user_sorting_options = ['first_name', 'last_name', 'email', 'date_account_created']
+        sender_sorting_options = list(map(lambda s: 'sender_'+s, user_sorting_options)) # sender_first_name, sender_last_name, etc...
```

Comment:
Python: "There should only be one obvious way to do everything"
Also python: "`list(map(lambda i: bool(i), a)) == [bool(i) for i in a]`"
lollllll